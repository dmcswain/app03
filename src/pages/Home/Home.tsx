import { useEffect } from 'react';
import { Box, Typography } from '@mui/material';
import { getUsers } from 'db';
import usePersistedState from 'hooks/usePersistedState';
import { useDispatch, useStore } from 'store/Provider';

export interface HomeProps {}

const Home: React.FC<HomeProps> = () => {
   const { currentUser } = useStore();
   const dispatch = useDispatch();
   const [lastLoginStatus, setLastLoginStatus] = usePersistedState('out');

   useEffect(() => {
      if (currentUser) {
         //@ts-ignore
         setLastLoginStatus('in');
      } else {
         //@ts-ignore
         setLastLoginStatus('out');
      }
   }, [currentUser, setLastLoginStatus]);

   useEffect(() => {
      if (currentUser) return;
      if (!getUsers()) return;
      if (lastLoginStatus === 'out') return;

      const users = getUsers();
      const lastUser = users![users!.length - 1];

      dispatch({ type: 'login', payload: lastUser });
      // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [dispatch]);

   return (
      <Box
         sx={{
            display: 'flex',
            height: '80vh',
            width: 'inherit',
            alignItems: 'center',
            justifyContent: 'center',
         }}
      >
         <Typography align='center' variant='h2'>
            Home
         </Typography>
      </Box>
   );
};

export default Home;
