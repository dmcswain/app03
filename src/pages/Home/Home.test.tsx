import { render, screen } from '@testing-library/react';
import Home, { HomeProps } from './Home';

describe('Home', () => {
    const defaultProps: HomeProps = {};

    it('should render', () => {
        const props = { ...defaultProps };
        const { asFragment } = render(<Home {...props} />);

        expect(asFragment()).toMatchSnapshot();
        expect(screen.getByText('Home')).toBeTruthy();
    });
});
