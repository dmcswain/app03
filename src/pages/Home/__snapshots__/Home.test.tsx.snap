// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-ovbz0g"
  >
    <h2
      class="MuiTypography-root MuiTypography-h2 MuiTypography-alignCenter css-icizyq-MuiTypography-root"
    >
      Home
    </h2>
  </div>
</DocumentFragment>
`;
